app/controllers/pieces_controller.rb

```diff
@@ -1,11 +1,14 @@
 class PiecesController < ApplicationController
-  def show
-    @piece = Piece.find(params[:id])
-  end
-
   def update
-    current_piece.update_attributes(pieces_params)
-    redirect_to game_path(current_piece.game)
+    row = pieces_params[:row].to_i
+    column = pieces_params[:column].to_i
+
+    if current_piece.valid_move?(row, column)
+      current_piece.update_attributes(pieces_params)
+      render text: 'updated!'
+    else
+      render text: 'invalid', status: :bad_request
```

Comment:
Yes, we'll be calling `move_to!` here.

If you want to add it now. Go for it. Or it could be a separate pull request.

Re: http status.

This should also work

```Ruby
## For good
render status: :ok

## For bad
render status: :bad_request
```